cmd/soroban-rpc/internal/methods/get_transactions.go

```diff
@@ -0,0 +1,230 @@
+package methods
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io"
+
+	"github.com/creachadair/jrpc2"
+	"github.com/creachadair/jrpc2/handler"
+	"github.com/stellar/go/ingest"
+	"github.com/stellar/go/support/errors"
+	"github.com/stellar/go/support/log"
+	"github.com/stellar/go/toid"
+
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/db"
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/ledgerbucketwindow"
+)
+
+// TransactionsPaginationOptions defines the available options for paginating through transactions.
+type TransactionsPaginationOptions struct {
+	Cursor *toid.ID `json:"cursor,omitempty"`
+	Limit  uint     `json:"limit,omitempty"`
+}
+
+// GetTransactionsRequest represents the request parameters for fetching transactions within a range of ledgers.
+type GetTransactionsRequest struct {
+	StartLedger uint32                         `json:"startLedger"`
+	Pagination  *TransactionsPaginationOptions `json:"pagination,omitempty"`
+}
+
+// isValid checks the validity of the request parameters.
+func (req GetTransactionsRequest) isValid(maxLimit uint, ledgerRange ledgerbucketwindow.LedgerRange) error {
+	if req.Pagination != nil && req.Pagination.Cursor != nil {
+		if req.StartLedger != 0 {
+			return errors.New("startLedger and cursor cannot both be set")
+		}
+	} else if req.StartLedger < ledgerRange.FirstLedger.Sequence || req.StartLedger > ledgerRange.LastLedger.Sequence {
+		return errors.Errorf("start ledger must be between the oldest ledger: %d and the latest ledger: %d for this rpc instance.", ledgerRange.FirstLedger.Sequence, ledgerRange.LastLedger.Sequence)
+	}
+
+	if req.Pagination != nil && req.Pagination.Limit > maxLimit {
+		return fmt.Errorf("limit must not exceed %d", maxLimit)
+	}
+
+	return nil
+}
+
+type transactionInfo struct {
+	// Successful indicates whether the transaction was successful or not
+	Successful bool `json:"status"`
+	// ApplicationOrder is the index of the transaction among all the transactions
+	// for that ledger.
+	ApplicationOrder int32 `json:"applicationOrder,omitempty"`
+	// FeeBump indicates whether the transaction is a feebump transaction
+	FeeBump bool `json:"feeBump,omitempty"`
```

Comment:
what would `omitempty` mean in this context?

Reply:
Copied it from getTransaction response but did not realise the omitempty tags were present. Except the diagnosticEventsXDR, it is not required here